src/main/java/arthur/Parser.java

```diff
@@ -0,0 +1,77 @@
+package arthur;
+
+/**
+ * Handles decoding of the user commands
+ */
+public class Parser {
+    private String command;
+    private boolean isEnd;
+
+    /**
+     * Constructor for the parser object
+     * @param inst String instruction to be followed
+     */
+    public Parser(String inst) {
+        this.command = inst;
+    }
+
+    /**
+     * Uses the command string to determine appropriate action.
+     * If input is "list", will execute listOut method. If not, adds
+     * instruction to tasks list.
+     * @param taskList The tasklist of tasks and operations
+     * @param storage The object to access data file in storage
+     * @param ui The arthur.Ui class to use for user interaction
+     */
+    public void execute(TaskList taskList, Storage storage, Ui ui) {
+        String[] temp = this.command.split(" ", 2);    // Gets the first word

```

Comment:
Instead of `temp`, maybe you can name the variable as `firstWord`, which is more intuitive

Reply:
Thanks for the suggestion. That is a bit better.